### PR TITLE
Fixed error message returned by regex on field and index names

### LINKED ISSUE #1972: Index and field names with leading numbers fail validation
### What's going wrong?

When creating a new index or field with a leading number Pilosa returns the following error, which implies leading numbers are valid:

`{"success":false,"error":{"message":"invalid index or field name, must match [a-z0-9_-]"}}`

### What was expected?

Index and field names with leading numbers are allowed or a different regex pattern should be returned in the error message.

### Steps to reproduce the behavior

Create an index or field using curl:

`curl 'localhost:10101/index/2test' -X POST`

`curl 'localhost:10101/index/test/field/3ddsfs' -X POST`

### Information about your environment (OS/architecture, CPU, RAM, cluster/solo, configuration, etc.)

Version 1.2.0

Solo server on Docker in MacOS

### PR DESCRIPTION
**Pull request by hackskills**
## Overview

This PR fixes the error message regex returns on field and index names with leading digits.

Fixes #1972 

## Pull request checklist

- [x] I have read the [contributing guide](https://github.com/pilosa/pilosa/blob/master/CONTRIBUTING.md).
- [x] I have agreed to the [Contributor License Agreement](https://cla-assistant.io/pilosa/pilosa).
- [x] I have updated the [documentation](https://github.com/pilosa/pilosa/tree/master/docs).
- [ ] I have resolved any merge conflicts.
- [ ] I have included tests that cover my changes.
- [x] All new and existing tests pass.
- [x] Make sure PR title conforms to convention in CHANGELOG.md.
- [ ] Add appropriate changelog label to PR (if applicable).

## Code review checklist
This is the checklist that the reviewer will follow while reviewing your pull request. You do not need to do anything with this checklist, but be aware of what the reviewer will be looking for.

- [ ] Ensure that any changes to external docs have been included in this pull request.
- [ ] If the changes require that minor/major versions need to be updated, tag the PR appropriately.
- [ ] Ensure the new code is [properly commented](https://github.com/golang/go/wiki/CodeReviewComments#doc-comments) and follows [Idiomatic Go](https://dmitri.shuralyov.com/idiomatic-go).
- [ ] Check that tests have been written and that they cover the new functionality.
- [ ] Run tests and ensure they pass.
- [ ] Build and run the code, performing any applicable integration testing.
- [ ] Make sure PR title conforms to convention in CHANGELOG.md.
- [ ] Make sure PR is tagged with appropriate changelog label.

### CODE REVIEW

File: docs/getting-started.md [hunk 1]
```diff
@@ -63,6 +63,8 @@ curl localhost:10101/index/repository -X POST
 ``` response
 {"success":true}
 ```
+Note:
```
Comment (jaffee) [inline docs/getting-started.md]: I know we do the `Note:` thing above, but the newline after it doesn't actually get rendered. I say we drop the `Note:` line OR if we actually do want a fancy note block, there is an example of that above where we do `<div class="note">`

File: docs/getting-started.md [hunk 2]
```diff
@@ -63,6 +63,8 @@ curl localhost:10101/index/repository -X POST
 ``` response
 {"success":true}
 ```
+Note: 
+The index name must be 64 characters or less in length and must  be a lower case string: a lowercase letter followed by alphanumeric characters and/or `_-`.
```
Comment (jaffee) [inline docs/getting-started.md]: This is a bit wordy and uses both "lowercase" and "lower case" (lowercase seems to be correct, though I had to look it up)

Maybe:
"The index name must be 64 characters or less, start with a letter, and consist only of lowercase alphanumeric characters or `_-`."

File: pilosa.go [hunk 3]
```diff
@@ -48,7 +48,7 @@ var (
 	ErrInvalidView      = errors.New("invalid view")
 	ErrInvalidCacheType = errors.New("invalid cache type")
 
-	ErrName  = errors.New("invalid index or field name, must match [a-z0-9_-]")
+	ErrName  = errors.New("invalid index or field name, must match [a-z][a-z0-9_-] and contain at most 64 characters")
```
Comment (jaffee) [inline pilosa.go]: I think `[a-z][a-z0-9_-]*`

File: go.sum [hunk 4]
```diff
@@ -145,6 +145,7 @@ golang.org/x/text v0.3.2/go.mod h1:bEr9sfX3Q8Zfm5fL9x+3itogRgK3+ptLWKqgva+5dAk=
 golang.org/x/tools v0.0.0-20180917221912-90fa682c2a6e/go.mod h1:n7NCudcB/nEzxVGmLbDWY5pfWTLqBcC2KZ6jyYvM4mQ=
 golang.org/x/tools v0.0.0-20190429231329-9d4d845e86f1 h1:MSSXVSCgrxTAYytvleklMKlLdxjexiJWNffJciO1nCI=
 golang.org/x/tools v0.0.0-20190429231329-9d4d845e86f1/go.mod h1:RgjU9mgBXZiqYHBnxXauZ1Gv1EHHAz9KjViQ78xBX0Q=
+gopkg.in/check.v1 v0.0.0-20161208181325-20d25e280405 h1:yhCVgyC4o1eVCa2tZl7eS0r+SDo693bJlVdllGtEeKM=
```
Comment (jaffee) [inline go.sum]: I'm not sure why this is happening...

File: docs/getting-started.md [hunk 5]
```diff
@@ -63,6 +63,7 @@ curl localhost:10101/index/repository -X POST
 ``` response
 {"success":true}
 ```
+The index name must be 64 characters or less, start with a letter, and consist only of lowercase alphanumeric characters or _-.
```
Comment (jaffee) [inline docs/getting-started.md]: I think it might still need the back ticks around the underscore Dash

Comment (hackskills) [reply docs/getting-started.md]: Whoops my bad
